### ISSUE TITLE
Integrate `ConfigurationPassImpl` into the configuration loading

### ISSUE THREAD

**Issue by shonfeder** [informalsystems/apalache#2066]
Currently the `ConfigurationPassImpl` does some gnarly things by mutating the state of the options map based on what it reads out of the tla `cfg` files. Instead, we should add a stage in program configuration that reads all needed config data from a `cfg` (if given) and merges that into the program configuration.

This will be required for #1174 and to make the various procedures thread safe. Moreover, this will enable us to load all the data that usually goes into a TLA `cfg` file via other config sources, like a JSON blob or a HOCON file.

**Comment by shonfeder** [comment]
This is tougher than I realized:

When deriving init and next from a TLC config, then can be specified in two different ways:

  1. Explicitly by `INIT` and `NEXT` configs.
  2. Implicitly by a `SPECIFICATION`, which is just the name of a specification operator in the module itself.

In order to determine `Init` and `Next` in this case, we need to parse the module and extract the named "specification", then inspect it to discover the implied next and init predicates.  So these cannot be derived in this until after we've parsed the module.

In other words, the values for the next and init preds are not actually "options" from this point on, but are instead dynamically determined values.

The best approach here might be as follows:

- Replace the `init` and `next` option fields with a sum type like `TransitionPreds = InitNext of string * string | Specification of string`  
- Introduce a special purpose mutable class that can track the state of the "init" and "next" values once those are finally determined post-parsing. The default of course being that we fall back to the configured value.
